qlib/kernel/loader/vdso.rs

```diff
@@ -39,6 +42,83 @@ pub struct VdsoInternal {
     pub vdsoAddr: u64,
     pub vdsoLen: usize,
     pub phdrs: Vec<ProgramHeader64>,
+    pub vdso_symbols: Option<VdsoArchSymbols>,
+}
+
+struct VdsoSymbol {
+    name:&'static str,
+    page_offset: Option<u64>,
+}
+
+/// Bookkeep the exported symbols for x86_64.
+#[cfg(target_arch = "x86_64")]
+pub struct VdsoArchSymbols {
+}
+
+#[cfg(target_arch = "x86_64")]
+impl VdsoArchSymbols {
+    pub fn set_symbols_page_offset(&mut self, elf_file: &ElfFile, symbol_table: &[Entry64]) {
+        todo!();
+    }
+
+    pub fn get_symbol_page_offset(&self, name: &str) -> Option<u64> {
+       todo!();
+       None
+    }
+}
+
+/// Bookkeep exported symbols for aarch64.
+#[cfg(target_arch = "aarch64")]
+pub struct VdsoArchSymbols {
+    symbols:[VdsoSymbol; Self::TOTAL_SYMBOLS],
+}
+
+#[cfg(target_arch = "aarch64")]
+impl Default for VdsoArchSymbols {
+    fn default() -> Self {
+        let _symbols: [VdsoSymbol; Self::TOTAL_SYMBOLS] = [
+            VdsoSymbol {name: "__kernel_gettimeofday", page_offset: None},
+            VdsoSymbol {name: "__kernel_clock_getres", page_offset: None},
+            VdsoSymbol {name: "__kernel_rt_sigreturn", page_offset: None},
+            VdsoSymbol {name: "__kernel_clock_gettime", page_offset: None}];
+        Self {
+            symbols: _symbols,
+        }
+    }
+}
+
+#[cfg(target_arch = "aarch64")]
+impl VdsoArchSymbols {
+    const TOTAL_SYMBOLS: usize = 4;
+
+    pub fn set_symbols_page_offset(&mut self, elf_file: &ElfFile, symbol_table: &[Entry64]) {
+        let mut found_symbols = 0;
+        for sym in 0..Self::TOTAL_SYMBOLS {
```

Comment:
qkernel already uses the `hashbrown`, a KV store would make life easier here

Reply:
IMO, too much underling complexity for such a minimal case...

Reply:
I don't see the complexity of using a hashmap, when the alternative is a nested  for loop with strcmp

Reply:
Because it is done only once, for an unnoticeable small number of elements, with unnoticeable short names as keys and very simple logic for to mess it up.

Reply:
nit: not just once. Every time a signal happens the kernel for loop through the list and strcmp up to 4 times to find the sigreturn symbol. Not a big difference with only 4 elements though.

Reply:
> nit: not just once.

Sorry, I didn't specify that I meant storing.

> up to 4 times to find the sigreturn symbol.

As we are (for now) only concerned with sigreturn,
it could be placed at [0].

... but again:
> Not a big difference with only 4 elements though.

P.S. I just learned what 'nit' means :)